### PR TITLE
Tooling update

### PR DESCRIPTION
**Pull request by arnau**
This pull request is for you to consider whether you'd like to move from `.ruby-version` to `.tool-versions` a-la [asdf](https://asdf-vm.com/) such that we can pin both Ruby and Nodejs runtimes.

If the above feels like the right move, then we should consider moving to Ruby 3 and Node 17 (latest versions as shown in the `.tool-versions` although we could relax the pinning a bit.

### CODE REVIEW

**Review by jamietanna — COMMENTED** [review]
This sounds like a reasonable solution :+1:

We'll need to amend [GitHub Actions](https://github.com/alphagov/data-standards-authority/blob/f17319c6590d3ecc8727629a041f10e4f4592504/.github/workflows/ci.yml#L19-L27) and our Docker config to get this working, but I'm happy with this as a path forward :+1:

**Comment by arnau** [comment]
@jamietanna @charlesbaird-dsa I added the Nix and Niv config files in case you want to test it.

**Comment by arnau** [comment]
Ignore it for now, installing nokogiri is turning into the classic nightmare.

File: standards-catalogue/nix/sources.nix [hunk 1]
```diff
@@ -0,0 +1,174 @@
+# This file has been generated by Niv.
```
Comment (jamietanna) [inline standards-catalogue/nix/sources.nix:1]: :question: should this be commited?

Comment (arnau) [reply standards-catalogue/nix/sources.nix:1]: Yes, this is what Niv uses to pin down the version of nixpkgs which will have stronger guarantees than just plain nix

**Pull request closed without merging**